pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -625,6 +647,150 @@ func (pc *PodController) initializeSecondaryInterfaceStore() error {
+		podNamespace := ifconf.ContainerInterfaceConfig.PodNamespace
+		podName := ifconf.ContainerInterfaceConfig.PodName
+		podRef := klog.KRef(podNamespace, podName)
+		pod, err := pc.podLister.Pods(podNamespace).Get(podName)
+		if err != nil {
+			klog.ErrorS(err, "Failed to get Pod", "Pod", podRef)
+			continue
+		}
+		_, found := checkForPodSecondaryNetworkAttachment(pod)
+		if !found {
+			klog.V(2).InfoS("Pod does not have a NetworkAttachmentDefinition", "Pod", podRef)
+			continue
+		}
+		netStatus, err := netdefutils.GetNetworkStatus(pod)
+		if err != nil {
+			klog.ErrorS(err, "Failed to get NetworkStatus for Pod", "Pod", podRef)
+			continue
+		}
+
+		netObj := pod.Annotations[netdefv1.NetworkAttachmentAnnot]
+		// Parse Pod annotation and proceed with the secondary network configuration.
+		networkList, err := netdefutils.ParseNetworkAnnotation(netObj, pod.Namespace)
+		if err != nil {
+			klog.ErrorS(err, "Error when parsing network annotation", "annotation", netObj)
+			continue
+		}
+
+		cache, err := pc.buildVFDeviceIDListPerPod(podName, podNamespace)
+		if err != nil {
+			return err
+		}
+
+		var sriovInfo []podSriovVFDeviceIDInfo
+		for _, status := range netStatus {
+			if status.DeviceInfo == nil {
+				continue
+			}
+			resourceName, err := getNADResourceName(pod, status.Name, allNADs, networkList)
+			if err != nil {
+				klog.ErrorS(err, "Failed to get resourceName", "Pod", podRef, "network", status.Name)
+				continue
+			}
+			for _, vfDeviceInfo := range cache {
+				if vfDeviceInfo.resourceName == resourceName && vfDeviceInfo.vfDeviceID == status.DeviceInfo.Pci.PciAddress {
+					vf := podSriovVFDeviceIDInfo{
+						resourceName: resourceName,
+						vfDeviceID:   vfDeviceInfo.vfDeviceID,
+						ifName:       status.Interface,
+					}
+					sriovInfo = append(sriovInfo, vf)
+					// Add the interface to the Secondary interfaceStore.
+					containerMAC, _ := net.ParseMAC(status.Mac)
+					secondaryInterfaceConfig := interfacestore.NewContainerInterface(
+						vfDeviceInfo.vfDeviceID,
+						ifconf.ContainerInterfaceConfig.ContainerID,
+						podName,
+						podNamespace,
+						status.Interface,
+						ifconf.ContainerInterfaceConfig.NetNS,
+						containerMAC,
+						parseIPs(status.IPs),
+						0)
+					klog.InfoS("Adding secondary interface to interfaceStore", "Pod", podRef, "interface", status.Interface)
+					pc.interfaceStore.AddInterface(secondaryInterfaceConfig)
+				}
+			}
+		}
+		podKey := podKeyGet(podName, podNamespace)
+		pc.vfDeviceIDUsageMap.Store(podKey, sriovInfo)
+	}
+	klog.InfoS("Successfully initialized the secondary interface store for SR-IOV devices")
+	return nil
+}
+
+func getAllNADs(netAttachDefClient netdefclient.K8sCniCncfIoV1Interface) (map[k8stypes.NamespacedName]*netdefv1.NetworkAttachmentDefinition, error) {
+	nadList, err := netAttachDefClient.NetworkAttachmentDefinitions("").List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
```

Comment:
If user doesn't install the CRD, will this fail the initialization and affect VLAN type interfaces?

Reply:
Good catch, let me update the code to check the error if the CRD does not exist.

Reply:
Updated.